peps/pep-0705.rst

```diff
@@ -77,106 +82,297 @@ The problem disappears if we don't have mutator methods in ``Movie``. This could
+    album: Album = { name: "Flood", year: 1990 }
+    album["year"] = 1973  # OK
+    album["name"] = "Dark Side Of The Moon"  # Also OK now
 
-* For each key in ``A``, ``B`` has the corresponding key and the corresponding value type in ``B`` is consistent with the value type in ``A``.
+Subclasses can narrow value types of read-only entries::
 
-* For each required key in ``A``, the corresponding key is required in ``B``.
+    class AlbumCollection(TypedDict, readonly=True):
+        albums: Collection[Album]
+
+    class RecordShop(AlbumCollection, TypedDict):
+        name: str
+        albums: list[Album]
+
+Subclasses can also require keys that are read-only but not required in the superclass::
+
+    class OptionalName(TypedDict, readonly=True):
+        name: NotRequired[str]
+
+    class Person(OptionalName, TypedDict):
+        name: Required[str]
+
+    person: Person = {}  # Type check error: "name" required
+
+Note that these are just consequences of structural typing, but they are highlighted here as the behavior now differs from the rules specified in :pep:`589`.
+
+Type consistency
+----------------
+
+A TypedDict type with ``other_keys=False`` is consistent with ``Mapping[str, V]``, where ``V`` is the union of all its value types. For instance, the following type is consistent with ``Mapping[str, int | str]``::
+
+    class Person(TypedDict, other_keys=False):
+        name: str
+        age: int
+
+A TypedDict type ``A`` is consistent with TypedDict ``B`` if ``A`` is structurally compatible with ``B``. This is true if and only if all of the following are satisfied:
+
+* For each key in ``B``, ``A`` has the corresponding key and the corresponding value type in ``A`` is consistent with the value type in ``B``, unless the key in ``B`` is of type ``ReadOnly[NonRequired[Any]]``, in which case it may be missing in ``A`` provided ``A`` allows other keys.
+* For each non-read-only key in ``B``, the corresponding value type in ``B`` is also consistent with the corresponding value type in ``A``.
+* For each required key in ``B``, the corresponding key is required in ``A``.
+* For each non-read-only, non-required key in ``B``, the corresponding key is not required in ``A``.
+* If ``B`` does not allow other keys, then ``A`` does not allow other keys.
+* If ``B`` does not allow other keys, then for each key in ``A``, ``B`` has the corresponding key.
 
 Discussion:
 
-* Value types behave covariantly, since ``TypedMapping`` objects have no mutator methods. This is similar to container types such as ``Mapping``, and different from relationships between two ``TypedDict`` types. Example::
+* All non-specified keys in a type that allows other keys are implicitly of type ``ReadOnly[NotRequired[Any]]``.
+
+* Read-only value types behave covariantly, as they cannot be mutated. This is similar to container types such as ``Mapping``, and different from non-read-only value types, which behave invariantly. Example::
```

Comment:
```suggestion
* Read-only value types behave covariantly, as they cannot be mutated. This is similar to container types such as ``Sequence``, and different from non-read-only value types, which behave invariantly. Example::
```

``Mapping`` isn't a great example because it's invariant in the key type